src/sentry/models/monitor.py

```diff
@@ -162,8 +193,9 @@ def mark_failed(self, last_checkin=None):
 
         event_manager = EventManager(
             {
-                "logentry": {"message": "Monitor failure: %s" % (self.name,)},
-                "contexts": {"monitor": {"id": six.text_type(self.guid)}},
+                "logentry": {"message": "Monitor failure: %s (%s)" % (self.name, reason)},
+                "contexts": {"monitor": get_monitor_context(self)},
```

Comment:
Will the new monitor context need to be added in semaphore as well?

Reply:
Nope - contexts are freeform schema, and the only thing typing does is create known fields and tags.

Reply:
Ok. When we want to index/search on these in a more performant way we can formalize the interface more.